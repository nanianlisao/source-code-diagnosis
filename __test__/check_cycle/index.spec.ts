import { expect, test } from 'vitest'
import { fileURLToPath } from "node:url";
import path, { dirname } from "node:path";
import { checkDependents,initLogger  , checkDetectCycle ,checkDependencies} from '../../index.js'

const __filename = fileURLToPath(import.meta.url);

test('Detect circular dependencies in the specified directory.', () => {
  const cwd = path.resolve(dirname(__filename),"features","cycle");  
  const response = checkDetectCycle({
    cwd,
  })


  for (const item of response) {
    expect(item.length).toBe(2)
  }
  

  expect(response.length).toBe(1)
})