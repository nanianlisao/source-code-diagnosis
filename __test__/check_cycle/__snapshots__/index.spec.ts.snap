// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Detect circular dependencies in the specified directory. 1`] = `"a.js -> b.js , b.js -> a.js :: a.js -> b.js , b.js -> c.js , c.js -> a.js :: a.js -> c.js , c.js -> a.js"`;

exports[`Ignore node_modules and dist directories. 1`] = `""`;
